src/lib.rs

```diff
@@ -65,6 +70,17 @@ pub const SIG_SIZE: usize = 96;
 
 const ERR_OS_RNG: &str = "could not initialize the OS random number generator";
 
+#[cfg(feature = "codec-support")]
+impl_codec_for!(PublicKey);
+#[cfg(feature = "codec-support")]
+impl_codec_for!(Signature);
+#[cfg(feature = "codec-support")]
+impl_codec_for!(DecryptionShare);
+#[cfg(feature = "codec-support")]
+impl_codec_for!(PublicKeySet);
+#[cfg(feature = "codec-support")]
+impl_codec_for!(Ciphertext);
```

Comment:
Let's move those macro calls to `codec_impl`, then we don't need five `#[cfg(feature = "codec-support")]`.